code/parse/sexp/LuaSEXP.cpp

```diff
@@ -526,29 +526,75 @@ void LuaSEXP::parseTable() {
 		}
 
 		if ((strcmp(type.first.c_str(), "enum")) == 0) {
-			type.second = increment_enum_list_id();
+			
 			required_string("+Enum Name:");
 
 			SCP_string enum_name;
 			stuff_string(enum_name, F_NAME);
 
 			dynamic_sexp_enum_list thisList;
-			thisList.name = enum_name;
+			int list_position = get_dynamic_enum_position(enum_name);
+			bool new_enum = false;
+
+			if (list_position >= 0) {
+				type.second = list_position + (int)First_available_opf_id;
+
+				//Do this just in case we're going to use this list as a template
+				thisList.name = Dynamic_enums[list_position].name; 
+				for (const SCP_string& enum_item : Dynamic_enums[list_position].list) {
+					thisList.list.push_back(enum_item);
+				}
+			} else {
+				new_enum = true;
+				thisList.name = enum_name;
+			}
 
-			SCP_vector<SCP_string> list_items;
 			while (optional_string("+Enum:")) {
+				new_enum = true;
+
+				//If we're making a new Enum based off another one, let's give it a unique name
+				if (list_position >= 0) {
+					char newName[NAME_LENGTH];
+					strcpy_s(newName, enum_name.c_str());
+					strcat(newName, std::to_string(list_position).c_str());
```

Comment:
Hmm. That's a good question. My thinking was that pre-parsed Enums would be the templates and anything that's created inline like this that uses a previously existing enum would be a modifier of that template. This method allows to use the same syntax for creating a new Enum as well as modifying. I wasn't really intending for these inline edits to be further modified.

So you could create an Enum at the top that's something like

> +Enum Name: ShipClasses
> +Enum: Apollo
> +Enum: Hercules
> +Enum: Athena

and the inline you have something like

> *Sexp 1*
> +Enum Name: ShipClasses //this sexp uses all three listed Ship Classes
> 
> *Sexp 2*
> +Enum Name: ShipClasses //this sexp uses all three listed Ship Classes
> +Enum: Ursa //but can also work on the Ursa
>
> *Sexp 3*
> +Enum Name: ShipClasses //this sexp uses all three listed Ship Classes
> +Enum: Valkyrie //but can also work on the Valkyrie
> +Enum: Ursa //and the Ursa

Reply:
Also not sure why I did that bit as a char array instead of a string since the destination is a string anyway. So I just redid the same code as a string and that should remove the risk of overflow.

Reply:
Fair, I guess having the pre parse enums as templates should suffice then